Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -7441,28 +7348,28 @@ RefPtr<CSSValue> consumeJustifyItems(CSSParserTokenRange& range)
     if (identMatches<CSSValueAuto>(range.peek().id()))
         return nullptr;
     CSSParserTokenRange rangeCopy = range;
-    RefPtr<CSSPrimitiveValue> legacy = consumeIdent<CSSValueLegacy>(rangeCopy);
-    RefPtr<CSSPrimitiveValue> positionKeyword = consumeIdent<CSSValueCenter, CSSValueLeft, CSSValueRight>(rangeCopy);
+    auto legacy = consumeIdent<CSSValueLegacy>(rangeCopy);
+    auto positionKeyword = consumeIdent<CSSValueCenter, CSSValueLeft, CSSValueRight>(rangeCopy);
     if (!legacy)
         legacy = consumeIdent<CSSValueLegacy>(rangeCopy);
     if (legacy) {
         range = rangeCopy;
         if (positionKeyword)
-            return createPrimitiveValuePair(legacy.releaseNonNull(), positionKeyword.releaseNonNull(), Pair::IdenticalValueEncoding::Coalesce);
+            return CSSValuePair::create(legacy.releaseNonNull(), positionKeyword.releaseNonNull());
```

Comment:
Looks like there are a few cases where the we are creating a CSSValuePair for a pair of CSSValueIDs wrapped in a CSSPrimitiveValue. Might be worth considering a CSSValueIDPair (for lack of a better name) that can avoid the primitive value overhead.

Reply:
Yes, makes sense to me.

Reply:
I could even do that when I make the `CSSIDValue` class or whatever the right name ends up being for it; won’t need any additional data members because there are enough free bits in `CSSValue` to store a value ID.